core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -363,6 +371,70 @@ public boolean isWaitForReady() {
   }
 
   /**
+   * Sets the maximum allowed message size acceptable from the remote peer.  If unset, this will
+   * default to the value set on the {@link ManagedChannel}.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/FIXME")
+  public CallOptions withMaxInboundMessageSize(int maxSize) {
+    checkArgument(maxSize >= 0, "invalid maxsize %s", maxSize);
+    CallOptions newOptions = new CallOptions(this);
+    newOptions.maxInboundMessageSize = maxSize;
+    return newOptions;
+  }
+
+  /**
+   * Sets the maximum allowed message size acceptable sent to the remote peer.  If unset, this will
+   * default to the value set on the {@link ManagedChannel}.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/FIXME")
+  public CallOptions withMaxOutboundMessageSize(int maxSize) {
+    checkArgument(maxSize >= 0, "invalid maxsize %s", maxSize);
+    CallOptions newOptions = new CallOptions(this);
+    newOptions.maxOutboundMessageSize = maxSize;
+    return newOptions;
+  }
+
+  /**
+   * Clears the maximum allowed message size acceptable from the remote peer.  The value set on the
+   * {@link ManagedChannel} may still be enforced.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/FIXME")
+  public CallOptions withoutMaxInboundMessageSize() {
```

Comment:
> Using a null to clear the value forces boxing. This currently does box, but that isn't part of the API; we could change it later to not box.

Why not make boxing part of the API?

> This is also slightly different than other APIs, because if the call option is not set, the channel default comes into play. I don't think any of the other call options behave this way.

`withAuthority()` and `withExecutor()` work this way too. Although `setExecutor()` doesn't have `@Nullable`, it doesn't `checkNotNull` either, so it seems reasonable to add `@Nullable`.